src/BooNEHadronQuasiElasticModel.cc

```diff
@@ -103,10 +103,14 @@ BooNEHadronQuasiElasticModel::ApplyYourself(const  G4HadProjectile  &aHadron,
   // Define the secondaries necessary for the final state
   // first set final state of hadron, now back to MeV 
   // Kinetic energy
-  theParticleChange.SetEnergyChange( (hadronOutMomentum.e() - hadrMass) * CLHEP::GeV);
-
+  // theParticleChange.SetEnergyChange( (hadronOutMomentum.e() - hadrMass) * CLHEP::GeV);
   // Direction
-  theParticleChange.SetMomentumChange(hadronOutMomentum.vect().unit());
+  // theParticleChange.SetMomentumChange(hadronOutMomentum.vect().unit());
+  
+  // ^^^^^^^ Deprecated, create a new dynamic particle instead ^^^^^^^
+  G4DynamicParticle * outHadron = 
+    new G4DynamicParticle(theHadron->GetDefinition(), hadronOutMomentum.vect() * CLHEP::GeV);
```

Comment:
Introduction of `new` here makes me worry about a possible memleak. Have you run this at production scales?

Reply:
Gazing at `BooNEHadronElasticProcess` and `BooNEHadronInelasticProcess`, there are instances of `new` in the constructor (and sure enough, these leak, but only once in a run), and of instances of `NuBeamTrackInformation`. This makes an entire `G4DynamicParticle`, which is larger. It will most likely leak, and it is unclear if this will cause problems -- if this runs okay for large numbers of events (100k? 1M?), then I guess we can live with this -- though I wonder if one can get away with making a reference, i.e.
```
theParticleChange.AddSecondary(&(G4DynamicParticle(G4Proton::Proton(), nucleonOutMomentum.vect() * CLHEP::GeV)));
```
...